main/main.cpp

```diff
@@ -323,147 +333,146 @@ void finalize_theme_db() {
-#ifdef TOOLS_ENABLED
-	OS::get_singleton()->print("  -e, --editor                      Start the editor instead of running the scene.\n");
-	OS::get_singleton()->print("  -p, --project-manager             Start the project manager, even if a project is auto-detected.\n");
-	OS::get_singleton()->print("  --debug-server <uri>              Start the editor debug server (<protocol>://<host/IP>[:<port>], e.g. tcp://127.0.0.1:6007)\n");
-#endif
-	OS::get_singleton()->print("  --quit                            Quit after the first iteration.\n");
-	OS::get_singleton()->print("  -l, --language <locale>           Use a specific locale (<locale> being a two-letter code).\n");
-	OS::get_singleton()->print("  --path <directory>                Path to a project (<directory> must contain a 'project.godot' file).\n");
-	OS::get_singleton()->print("  -u, --upwards                     Scan folders upwards for project.godot file.\n");
-	OS::get_singleton()->print("  --main-pack <file>                Path to a pack (.pck) file to load.\n");
-	OS::get_singleton()->print("  --render-thread <mode>            Render thread mode ['unsafe', 'safe', 'separate'].\n");
-	OS::get_singleton()->print("  --remote-fs <address>             Remote filesystem (<host/IP>[:<port>] address).\n");
-	OS::get_singleton()->print("  --remote-fs-password <password>   Password for remote filesystem.\n");
-
-	OS::get_singleton()->print("  --audio-driver <driver>           Audio driver [");
+// Build a comma separated list of audio drivers.
+// For example: 'CoreAudio', 'Dummy'.
+// Helper function for Main::print_help(const char *).
+String get_audio_drivers() {
+	Vector<String> drivers;
 	for (int i = 0; i < AudioDriverManager::get_driver_count(); i++) {
-		if (i > 0) {
-			OS::get_singleton()->print(", ");
-		}
-		OS::get_singleton()->print("'%s'", AudioDriverManager::get_driver(i)->get_name());
+		String audio_driver = AudioDriverManager::get_driver(i)->get_name();
+		drivers.push_back("'" + audio_driver + "'");
 	}
-	OS::get_singleton()->print("].\n");
+	return String(", ").join(drivers);
+}
 
-	OS::get_singleton()->print("  --display-driver <driver>         Display driver (and rendering driver) [");
+// Build a comma separated list of display & rendering drivers.
+// For example: 'macos' ('vulkan', 'opengl3'), 'headless' ('dummy').
+// Helper function for Main::print_help(const char *).
+String get_display_drivers() {
+	Vector<String> drivers;
 	for (int i = 0; i < DisplayServer::get_create_function_count(); i++) {
-		if (i > 0) {
-			OS::get_singleton()->print(", ");
-		}
-		OS::get_singleton()->print("'%s' (", DisplayServer::get_create_function_name(i));
-		Vector<String> rd = DisplayServer::get_create_function_rendering_drivers(i);
-		for (int j = 0; j < rd.size(); j++) {
-			if (j > 0) {
-				OS::get_singleton()->print(", ");
-			}
-			OS::get_singleton()->print("'%s'", rd[j].utf8().get_data());
-		}
-		OS::get_singleton()->print(")");
-	}
-	OS::get_singleton()->print("].\n");
```

Comment:
I extracted these two blocks into helper functions since the nested for loops made it difficult to quickly skim `print_help()`.

The helper functions are called `get_audio_drivers()` and `get_display_drivers()`.

Reply:
The GitHub code review diff is difficult to read since the for loops were moved out of the print_help() function. The two helper functions look like this:

```c++
String get_audio_drivers() {
    Vector<String> drivers;
    for (int i = 0; i < AudioDriverManager::get_driver_count(); i++) {
        String audio_driver = AudioDriverManager::get_driver(i)->get_name();
        drivers.push_back("'" + audio_driver + "'");
    }
    return String(", ").join(drivers);
}

String get_display_drivers() {
    Vector<String> drivers;
    for (int i = 0; i < DisplayServer::get_create_function_count(); i++) {
        String display_driver = DisplayServer::get_create_function_name(i);
        String rendering_drivers = "'" + String("', '").join(DisplayServer::get_create_function_rendering_drivers(i)) + "'";
        drivers.push_back("'" + display_driver + "' (" + rendering_drivers + ")");
    }
    return String(", ").join(drivers);
}
```

Reply:
The new helper function takes advantage of string literal concatenation to avoid having to write out `String(VERSION_NAME) +` and `+ String(VERSION_WEBSITE)`.

```
print_line(VERSION_NAME " v" + get_full_version_string() + " - " VERSION_WEBSITE);
```